simulation/behavior_tree_plugin/src/vehicle/follow_trajectory_sequence/follow_polyline_trajectory_action.cpp

```diff
@@ -0,0 +1,584 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <behavior_tree_plugin/vehicle/follow_trajectory_sequence/follow_polyline_trajectory_action.hpp>
+#include <scenario_simulator_exception/exception.hpp>
+
+#define LINE() \
+  std::cout << "\x1b[33m" __FILE__ "\x1b[31m:\x1b[36m" << __LINE__ << "\x1b[0m" << std::endl
+
+#define PRINT(...) std::cout << #__VA_ARGS__ " = " << std::boolalpha << (__VA_ARGS__) << std::endl
+
+namespace entity_behavior
+{
+namespace vehicle
+{
+auto FollowPolylineTrajectoryAction::calculateWaypoints()
+  -> const traffic_simulator_msgs::msg::WaypointsArray
+{
+  return traffic_simulator_msgs::msg::WaypointsArray();
+}
+
+auto FollowPolylineTrajectoryAction::calculateObstacle(
+  const traffic_simulator_msgs::msg::WaypointsArray &)
+  -> const std::optional<traffic_simulator_msgs::msg::Obstacle>
+{
+  return std::nullopt;
```

Comment:
[4c8ebcc](https://github.com/tier4/scenario_simulator_v2/pull/906/commits/4c8ebccad0fa2399b48869e2fde672f98f6bced6) fixes this.

Reply:
The return value of this function is currently only used for visualization, and is not "something that does not have to be implemented" but "something that must be implemented".
Currently the `calculateObstacle` function returns `std::nullopt` because obstacle avoidance is not implemented in FollowTrajectoryAction, and if obstacle avoidance is implemented, it is expected to be implemented in this location.